### ISSUE TITLE
DiscreteSlider do not allow a change of the name (e.g. change the language)

### ISSUE THREAD

**Issue by pat-schmitt** [holoviz/panel#1804]
panel 0.10.2

#### Description of expected behavior and the observed behavior
I would like to change the name of a DiscreteSlider according to the selected language, but I get  `TypeError: Constant parameter 'name' cannot be modified` .
This is already possible for most of the widgets (see example for IntSlider)

#### Complete, minimal, self-contained example code that reproduces the issue
This example shows how it is working for a IntSlider. To reproduce the error just uncomment the line `discrete_slider.name = languages[language_selector.value]`.

```
import panel as pn
pn.extension()

def set_language(arg=None):
    languages = {'en': 'english',
                 'de': 'deutsch'}

    int_slider.name = languages[language_selector.value]

    # discrete_slider.name = languages[language_selector.value]

language_selector = pn.widgets.RadioBoxGroup(name='select your language',
                                             options=['en', 'de'])
language_selector.param.watch(set_language, 'value')

int_slider = pn.widgets.IntSlider(start=1,
                                  end=9,
                                  step=1,
                                  value=1)

discrete_slider = pn.widgets.DiscreteSlider(options=[1, 3, 9],
                                            value=1)

set_language()

pn.Column(language_selector,
          int_slider,
          discrete_slider)

```

**Comment by philippjfr** [comment]
This appears to be fixed. Not sure quite when it was fixed but as of 0.12.4 it was okay.

**Comment by pat-schmitt** [comment]
Hi @philippjfr, thank you for your answer.

It is true that there is no `TypeError` anymore with panel 0.12.4, but it is still not working as expected. The problem is that it is possible to set a new `name` for the `DiscreteSlider`, but the name is not visually updated.

To make it clear here is a video of the behaviour of the example above (with uncommented line `discrete_slider.name = languages[language_selector.value]`):

https://user-images.githubusercontent.com/50843444/140878885-72faff1d-7e1e-46bd-b8bd-b97fb1f5a8a1.mp4

And here is a workaround with the above example showing the expected behaviour:

```
import panel as pn
pn.extension()


def set_language(arg=None):
    languages = {'en': 'english',
                 'de': 'deutsch'}

    int_slider.name = languages[language_selector.value]

    discrete_slider = pn.widgets.DiscreteSlider(options=[1, 3, 9],
                                                value=1,
                                                name=languages[language_selector.value])

    app.objects = [language_selector,
                   int_slider,
                   discrete_slider
                  ]


language_selector = pn.widgets.RadioBoxGroup(name='select your language',
                                             options=['en', 'de'])
language_selector.param.watch(set_language, 'value')

int_slider = pn.widgets.IntSlider(start=1,
                                  end=9,
                                  step=1,
                                  value=1)

discrete_slider = pn.widgets.DiscreteSlider(options=[1, 3, 9],
                                            value=1)

app = pn.Column(language_selector,
                int_slider,
                discrete_slider)

set_language()

app
```

https://user-images.githubusercontent.com/50843444/140879663-11423f3b-6dd9-4c97-aea2-800e632157d2.mp4

So the workaround is in principle to create a new `DiscreteSlider` whenever the language is changed.
